packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/object/controllers/ObjectDownloadController.java

```diff
@@ -0,0 +1,98 @@
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.object.controllers;
+
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IEditingContextSearchService;
+import org.eclipse.sirius.components.core.api.IObjectSearchService;
+import org.eclipse.sirius.web.application.object.services.api.IObjectExporter;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ContentDisposition;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/**
+ * The entry point of the HTTP API to download objects.
+ *
+ * @author gdaniel
+ */
+@Controller
+@RequestMapping("/api/editingcontexts/{editingContextId}/objects")
+public class ObjectDownloadController {
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final IObjectSearchService objectSearchService;
+
+    private final List<IObjectExporter> objectExporters;
+
+    public ObjectDownloadController(IEditingContextSearchService editingContextSearchService, IObjectSearchService objectSearchService, List<IObjectExporter> objectExporters) {
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.objectSearchService = Objects.requireNonNull(objectSearchService);
+        this.objectExporters = Objects.requireNonNull(objectExporters);
+    }
+
+    @ResponseBody
+    @GetMapping
+    public ResponseEntity<Resource> downloadObjects(@PathVariable String editingContextId, @RequestParam String contentType, @RequestParam List<String> objectIds) {
+
+        ResponseEntity<Resource> responseEntity = new ResponseEntity<>(null, new HttpHeaders(), HttpStatus.NOT_FOUND);
+
+        Optional<IEditingContext> optionalEditingContext = this.editingContextSearchService.findById(editingContextId);
```

Comment:
Why load a copy of the semantic data from the database instead of asking the editing context event processor registry? Was there a meaningful reason for this approach?

Reply:
> No, I just reused the same approach that is used in DocumentDownloadController

I forgot that we did it this way for this part of the code

> Should I update this to ask the editing context to the EditingContextEventProcessorRegistry as you suggest, or should I go the ObjectRestController way and dispatch a query?

Both are the same approach, you use the `EditingContextEventProcessorRegistry` to dispatch the query (thanks to the `IEditingContextDispatcher`).

If it's done this way already for the `DocumentDownloadController`, we can do it this way too here but we will have to keep in mind that reloading the entire editing context from scratch can have some performance impact. For example in SysON, this should involve reloading all the libraries which may not be instantaneous.

Reply:
I'll rebase it and merge it then